### PR TITLE
Added category & quiz page

### PR DESCRIPTION
**Pull request by sanket4120**
Files to review:

- actions (categoryActions, quizActions)
- reducers (categoryReducer, quizReducer)
- context (categoryContext, quizContext)
- pages (category, quiz)
- components (timer)
- utils (timerUtils)

### CODE REVIEW

File: src/pages/category/Category.js [hunk 1]
```diff
@@ -1,48 +1,65 @@
...
+  //     id: 2,
+  //     categoryName: 'technology',
+  //     image:
+  //       'https://res.cloudinary.com/svj/image/upload/v1652541674/technology_plfcs8.jpg',
+  //   },
+  //   {
+  //     id: 3,
+  //     categoryName: 'history',
+  //     image:
+  //       'https://res.cloudinary.com/svj/image/upload/v1652541674/history_pstz1r.jpg',
```
Comment (manojasarsa) [inline src/pages/category/Category.js:25]: you can remove this code, if no longer needed. Or can mention - "TODO" before it.

**Review by manojasarsa — APPROVED** [review]
The code is pretty organized and clean.
The category and quiz page functionality is working as expected.
You can remove the commented code.
Rest, it looks good!

File: src/actions/categoryActions.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  CATEGORY_SUCCESS,
+  CATEGORY_FAIL,
+} from '../constants/categoryConstants';
+
+export const getCategories = async (dispatch) => {
+  dispatch({ type: CATEGORY_REQUEST });
+  try {
+    const res = await axios.get('/api/category');
+    dispatch({ type: CATEGORY_SUCCESS, payload: res.data.categories });
+  } catch (e) {
```
Comment (manojasarsa) [inline src/actions/categoryActions.js:13]: You can make this as `catch(error)` to be consistent throughout the code. As you have used `catch(error)` everywhere.

File: src/actions/quizActions.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import axios from 'axios';
+import {
+  CURRENT_QUIZ_FAIL,
+  CURRENT_QUIZ_REQUEST,
+  CURRENT_QUIZ_SUCCESS,
+  QUIZ_FAIL,
+  QUIZ_REQUEST,
+  QUIZ_SUCCESS,
+} from '../constants/quizConstants';
```
Comment (manojasarsa) [inline src/actions/quizActions.js:9]: Nice way of extracting the quizConstants

**Review by simran61 — APPROVED** [review]
Looks good
Nice folder structure

**Pull request merged**